src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp

```diff
@@ -75,11 +79,49 @@ void handle_reshape::run(program& p) {
             // vector for storing nodes that are reorder type, for which splitted primitives are needed (except for the
             // first one where orginal reshape will be used)
             std::vector<program_node*> reorder_node_to_split;
+            std::vector<program_node*> onednn_users;
 
             // find the users of reshape that are reorder type, if none present then skip the current node
+            // find users who are onednn impl
             for (const auto& user : node->get_users()) {
                 if (user->is_type<reorder>())
                     reorder_node_to_split.push_back(user);
+                if (user->get_preferred_impl_type() == cldnn::impl_types::onednn)
+                    onednn_users.push_back(user);
+            }
+
+            // If onednn user doesn't support new input data type from future "reorder:_reshape_input_" reorder,
```

Comment:
[random spot] @wilson-seok Could you check if we still need reorder split at all? Maybe we can just remove that part and don't add this custom logic for onednn at all?

Reply:
@vladimir-paramuzov I'm checking performance regression impact of removing whole reorder split. There is no regression on gen9 platform. I'm checking on gen12lp. BTW you created new PR which touch reorder split in handle_reshape(https://github.com/openvinotoolkit/openvino/pull/16911). Do you think we will keep it? Or remove if there is no performance regression?

Reply:
If reorder split won't impact anything and can be removed, then handle_reshape modification in my PR can be discarded as well.

Reply:
Just finished performance regression check on gen12lp. There is no critical performance regression on both gen9 and gel12lp. I will update this PR to remove it. Thanks!

Reply:
Then I think it is better to let @jade-cho  to finalize his code. Partial implementation from this PR doesn't seem to be necessary. (especially because Jade's work is not complete yet)

Reply:
@wilson-seok I guess accuracy should be checked in the first place, as this code seem to be designed to avoid some issues with buffers propagation for optimized out primitives

Reply:
> @wilson-seok I guess accuracy should be checked in the first place, as this code seem to be designed to avoid some issues with buffers propagation for optimized out primitives

Let me try accuracy check for this PR.

Reply:
@vladimir-paramuzov Removing reorder split causes exec graph diff in 82 models. Accuracy check for them should take too long time. So why don't we keep it and add custom condition for onednn?

Reply:
Reverted to previous fix. It is hard to check accuracy impact within CF because 82 models are affected. Will create a ticket in backlog.